### PR TITLE
Bug 1821466 - Add nimbus message for fullscreen research surface

### PR DESCRIPTION
**Pull request by sarah541**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1821466

### CODE REVIEW

File: fenix/app/messaging.fml.yaml [hunk 1]
```diff
@@ -158,6 +158,13 @@ features:
                 - I_AM_NOT_DEFAULT_BROWSER
                 - DAY_3_AFTER_INSTALL
               action: MAKE_DEFAULT_BROWSER
+            viewpoint-invitation-message:
```
Comment (rocketsroger) [inline fenix/app/messaging.fml.yaml:161]: ⚠️ I don't think we should add these.  Should be testing only.

File: fenix/app/messaging.fml.yaml [hunk 2]
```diff
@@ -265,3 +272,5 @@ enums:
         description: A banner in the homescreen.
       notification:
         description: A local notification in the background, like a push notification.
+      fullscreen:
+        description: A fullscreen message.
```
Comment (rocketsroger) [inline fenix/app/messaging.fml.yaml:276]: 🔍 Let's have something more descriptive.  Maybe: "A full screen dialog that is intended to be disruptive."

File: fenix/app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -1180,6 +1180,13 @@
     <!-- Text B shown in the notification that pops up to re-engage the user -->
     <string name="notification_re_engagement_B_text">Find something nearby. Or discover something fun.</string>
 
+    <!-- Survey -->
+    <!-- Text shown in the fullscreen message that pops up to ask user to take a short survey.
+    The app name is in the text, due to limitations with localizing Nimbus experiments -->
+    <string name="nimbus_survey_message_text" tools:ignore="UnusedResources">Please help make Firefox better by taking a short survey.</string>
+    <!-- Preference for taking the short survey. -->
+    <string name="preferences_take_survey" tools:ignore="UnusedResources">Take Survey</string>
+
```
Comment (rocketsroger) [inline fenix/app/src/main/res/values/strings.xml:1189]: Should we add the string "No Thanks" as well?

**Review by rocketsroger — COMMENTED** [review]
Looks good.  Some questions.

**Review by rocketsroger — APPROVED** [review]
🚢

**Review by Amejia481 — APPROVED** [review]

**Pull request merged**